x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/physical/HashJoinExec.java

```diff
@@ -61,7 +58,8 @@ public void writeTo(PlanStreamOutput out) throws IOException {
         out.writePhysicalPlanNode(child());
         joinData.writeTo(out);
         out.writeNamedWriteableCollection(matchFields);
-        out.writeCollection(conditions, (o, v) -> v.writeTo(o));
+        out.writeNamedWriteableCollection(leftFields);
+        out.writeNamedWriteableCollection(rightFields);
```

Comment:
This is fine because you can't use this outside of a snapshot build and we don't offer wire compatibility between snapshot builds.